federation/ldap/src/main/java/org/keycloak/storage/ldap/mappers/membership/group/GroupLDAPStorageMapper.java

```diff
@@ -192,21 +194,37 @@ public String getStatus() {
             Set<String> visitedGroupIds = new HashSet<>();
 
             // Just add flat structure of groups with all groups at top-level
-            for (Map.Entry<String, LDAPObject> groupEntry : ldapGroupsMap.entrySet()) {
-                String groupName = groupEntry.getKey();
-                GroupModel kcExistingGroup = KeycloakModelUtils.findGroupByPath(realm, "/" + groupName);
-
-                if (kcExistingGroup != null) {
-                    updateAttributesOfKCGroup(kcExistingGroup, groupEntry.getValue());
-                    syncResult.increaseUpdated();
-                    visitedGroupIds.add(kcExistingGroup.getId());
-                } else {
-                    GroupModel kcGroup = realm.createGroup(groupName);
-                    updateAttributesOfKCGroup(kcGroup, groupEntry.getValue());
-                    realm.moveGroup(kcGroup, null);
-                    syncResult.increaseAdded();
-                    visitedGroupIds.add(kcGroup.getId());
-                }
+            final int GROUPS_PER_TRANSACTION = 1000;
```

Comment:
JFTR, this constant can be taken into the Keycloak configuration file as the parameter / setting, allowing to tune the count of records to be processed within one transaction, if desired / needed.